src/helpers.js

```diff
@@ -73,9 +73,35 @@ export function validateFind(directory: string, name: string | Array<string>) {
   }
 }
 
+const processMap: Map<string, Function> = new Map()
+
 export function wrapExec(callback: Function): Function {
-  return function(filePath: string, parameters: Array<string>, options: Object) {
-    return callback(filePath, parameters, Object.assign({ timeout: 10000 }, options)).catch(function(error) {
+  return function(filePath: string, parameters: Array<string>, options: Object = {}) {
+    let killed = false
+    const spawned = callback(filePath, parameters, Object.assign({ timeout: 10000 }, options))
+    let mirror = spawned
+
+    if (options.uniqueKey) {
+      if (typeof options.uniqueKey !== 'string') throw new Error('options.uniqueKey must be a string')
+
+      const oldValue = processMap.get(options.uniqueKey)
+      if (oldValue) {
```

Comment:
How about this? (The current setup works, this just makes it a bit clearer what's going on)

```js
if (processMap.has(options.uniqueKey)) {
  processMap.get(options.uniqueKey)()
}
```